### DISCUSSION TITLE
Round r after ø and ö

### DISCUSSION THREAD

**Discussion by sigurdhasle** [psb1558/Junicode-font#231] [General]
I have understood that the round r (ꝛ) should occur after letters that are round on the right-hand side; «oꝛ». When I select ss16 I get «oꝛ» – but «ør» and «ör». Should there not be an ꝛ after ø and ö, too?

![image](https://github.com/psb1558/Junicode-font/assets/47284316/acd596f3-ca61-4ea6-9ad8-60d31c381d66)

**Comment by psb1558** [comment]
Sounds reasonable.

**Reply by kenmcd** [reply]
@psb1558  
The description for ss16 says "Contextual r rotunda", but it appears to be a straight substitution.

![2023-08-19_17-00-25](https://github.com/psb1558/Junicode-font/assets/844920/ae6c9c55-96fe-4e25-bbeb-f37cf90bfa2f)

**Comment by psb1558** [comment]
It works in tandem with `calt`. If that is off for some reason, then you will get a straight substitution.

**Reply by kenmcd** [reply]
`calt` was On in my test (which was in FontCreator OpenType Designer).
Both their own shaper and HarfBuzz.
Will give it a try in LibreOffice tomorrow.

**Comment by psb1558** [comment]
It works every time for me.
![ss16](https://github.com/psb1558/Junicode-font/assets/16215767/7ca93d01-9b16-4a4a-9a68-d94aa60b5402)
See also the [specimen page](https://psb1558.github.io/Junicode-font/), where you can try it in all the browsers.

**Comment by sigurdhasle** [comment]
Hmm. When I select ss16 and deselect calt (or select r rotunda in the specimen page) I get round r-s everywhere.

![image](https://github.com/psb1558/Junicode-font/assets/47284316/511bd630-58a7-4207-aab8-110031001a5d)

**Comment by sigurdhasle** [comment]
… but when I select ss16 (or contextual r rotunda in the specimen page) I get round r-s after o – but not after ø or ö. I would like to have round r-s after ø and ö, too. I think that would be historically correct.

![image](https://github.com/psb1558/Junicode-font/assets/47284316/756e15ff-db0e-4d54-bad4-ba89d90c3d2d)

**Reply by psb1558** [reply]
I have done this as you requested, and it will be in the next release (whenever that appears).

**Comment by sigurdhasle** [comment]
Thank you very much! (And btw, thank you very much for an excellent, beautiful font!)

**Comment by kenmcd** [comment]
> It works every time for me.

The problem is in FontCreator.
I thought the alternate shaper is HarfBuzz, but it is not.
Apparently integrating HarfBuzz was discussed, but never happened.

> You can also choose to use a more advanced Shaping Engine, which behaves more like other software like Microsoft Word, Adobe InDesign, Serif Affinity Designer, and web browsers. To enable it, check the feature named “_shaper”. This will trigger (basic) normalization and will then process specific features in a fixed order, even if they are not selected, and after that it will process all remaining active features.

So it turns-out the alternate shaper is trying mimic Word - which apparently means mimicking the Word default of `calt` Off - or perhaps after other features based on this "fixed order".
I do not understand this, and there is only this very little explanation.

Conclusions: this feature works in Junicode as expected. 
FC has some dumb stuff in it.
Sorry for the confusion.
Please ignore me. 😀

**Comment by psb1558** [comment]
I will never ignore you.

That is very annoying about Font Creator (which I have never tried to use). HarfBuzz is *not* difficult to work with: I bet they expended more labor imitating the non-functionality of Word than they would have integrating HarfBuzz.

**Comment by kenmcd** [comment]
> That is very annoying about Font Creator (which I have never tried to use). HarfBuzz is _not_ difficult to work with: I bet they expended more labor imitating the non-functionality of Word than they would have integrating HarfBuzz.

Mostly I just use FontCreator OpenType Designer to view and test features - it is a lot more visual than FontLab (where you are just looking at code).
Clearly I have to be wary of the result - usually it is OK, but I have seen some issues before (and their forum has some shaper questions/issues)
Rarely make anything or even edit anything other than to do a test of some sort.
Use FontLab to actually make stuff.

Yeah, imitating Word? What does that mean? And why, instead of HarfBuzz?
So you are testing your font as if it is in an application with almost no OT features.
Quite odd. ???
But, I think it does remind us that this feature will not work in Word unless the user manually enables `calt` (which is Off by default in Word).
FontLab uses HarfBuzz; you can see the `harfbuzz.dll` in the program directory.

OK. Bitch session over. Should go do something useful.

**Comment by psb1558** [comment]
I have it constantly in mind that a huge set of users of any font will never use even its default features because MS Word does not have them on by default. I know this because my career as an English professor (which came, blissfully, to a conclusion last year) gave me a precise knowledge of how much smart but non-tech-savvy people know about font technology.

All but a few of them know nothing at all.

They give no thought to what goes into putting text on their screen. They will never use OT features (even ligatures) because they have no idea they exist. When they do know, they tend to view the simple act of enabling them in Word as an insurmountable challenge.

This is why I think the Word team's decision to leave default OT features *off by default* (no doubt thinking, "Well, folks can easily turn them on if they want them") an earthquake-magnitude disaster. Because a whole lot of heavy MS Word users *can't turn them on.*

End of my rant.

I generally start with [Font Goggles](https://fontgoggles.org/) for testing. Its feature controls are three-way buttons: default, on, off, with all set to "default" at the start of any session. It's running Harfbuzz, so you get a pretty realistic view of what a lot of apps are going to deliver. You can't use it to peer under the hood, but it tells you if you need to move on to other diagnostic tools.

**Reply by kenmcd** [reply]
> They will never use OT features (even ligatures) because they have no idea they exist.

Most fonts have no documentation at all. Absolutely nothing.
So a non-tech-savvy student has no chance of discovering the OT features even exist in Word.
Nothing in the Word interface gives any clues about any OT features.
And the always-fake-small-caps, and always-fake-subscripts-and-superscripts do not help understanding.
So I am not surprised your students were clueless.
And you are kinda at the other end of the spectrum, with far more knowledge then average. 

But even in the Affinity forum there are users with font questions which show they do not know what OpenType features are in their fonts, and/or they do not know how to use them.
Some of the more experienced graphic designers know about web sites like [FontDrop](https://fontdrop.info/) which at least lets them know what OpenType features are available in a particular font, but not everything each feature can do.
But nothing beats decent font documentation. (Then they do have to actually read it.)

> I generally start with Font Goggles for testing.

Yeah. I wish there was a Windows version. 
The GF folks talk about using it all the time.
I did install it in my macOS Ventura VM, and looked at it, and then had to go, and have not looked at it since.
Doing anything in my macOS VM is a slow giant PITA, so I tend to avoid it.

**Reply by psb1558** [reply]
I had no idea FontGoggles was Mac only. I wonder why: it is a packaged Python app, after all—you're supposed to be able to make it portable (my experience is that portability doesn't happen by itself, but you can certainly make it happen).

**Reply by jsbien** [reply]
> But nothing beats decent font documentation

What about creating it automatically? As far as I know, there is no appropriate tool, but cf. https://tug.org/TUGboat/tb44-1/tb136bien-unichart.html.

**Comment by kenmcd** [comment]
> I had no idea FontGoggles was Mac only. I wonder why: it is a packaged Python app, after all—you're supposed to be able to make it portable (my experience is that portability doesn't happen by itself, but you can certainly make it happen).

Yes. I looked doing it myself.
But IIRC the problem is they used an interface creation tool which is Mac-only.
If they had used something like Qt it could work.
Others have requested a Windows version in their tracker.
But, they say it would be a major project to convert it.
Again IIRC, at one time the GF manager wanted to do it (meaning and/or fund it).
But that idea seems to have since been dropped.

**Reply by psb1558** [reply]
I see. Vanilla is pretty popular with Glyphs people. It's a bit of an odd choice for a project like FontGoggles (not Glyphs-specific, generally useful), but I suppose it's tempting to go with an old familiar something.
